test/Common/HttpApiHandlerTests.cs

```diff
@@ -1,1665 +1,1857 @@
+        {
+            var httpApiHandler = new HttpApiHandler(GetTestExtension(), null);
+            var request = new HttpRequestMessage
+            {
+                RequestUri = new Uri(TestConstants.RequestUri),
+            };
+
+            var httpResponseMessage = httpApiHandler.CreateCheckStatusResponse(
+                request,
+                TestConstants.InstanceId,
+                new DurableClientAttribute
+                {
+                    TaskHub = TestConstants.TaskHub,
+                    ConnectionName = TestConstants.ConnectionName,
+                });
+
+            Assert.Equal(HttpStatusCode.Accepted, httpResponseMessage.StatusCode);
+            var content = httpResponseMessage.Content.ReadAsStringAsync().Result;
+            var status = JsonConvert.DeserializeObject<JObject>(content);
+            Assert.Equal(TestConstants.InstanceId, (string)status["id"]);
+            Assert.Equal(
+                $"{TestConstants.NotificationUrlBase}/instances/7b59154ae666471993659902ed0ba742?taskHub=SampleHubVS&connection=Storage&code=mykey",
+                (string)status["statusQueryGetUri"]);
+            Assert.Equal(
+                $"{TestConstants.NotificationUrlBase}/instances/7b59154ae666471993659902ed0ba742/raiseEvent/{{eventName}}?taskHub=SampleHubVS&connection=Storage&code=mykey",
+                (string)status["sendEventPostUri"]);
+            Assert.Equal(
+                $"{TestConstants.NotificationUrlBase}/instances/7b59154ae666471993659902ed0ba742/terminate?reason={{text}}&taskHub=SampleHubVS&connection=Storage&code=mykey",
+                (string)status["terminatePostUri"]);
+            Assert.Equal(
+                $"{TestConstants.NotificationUrlBase}/instances/7b59154ae666471993659902ed0ba742?taskHub=SampleHubVS&connection=Storage&code=mykey",
+                (string)status["purgeHistoryDeleteUri"]);
+            Assert.Equal(
+                $"{TestConstants.NotificationUrlBase}/instances/7b59154ae666471993659902ed0ba742/restart?taskHub=SampleHubVS&connection=Storage&code=mykey",
+                (string)status["restartPostUri"]);
+            Assert.Equal(
+                $"{TestConstants.NotificationUrlBase}/instances/7b59154ae666471993659902ed0ba742/suspend?reason={{text}}&taskHub=SampleHubVS&connection=Storage&code=mykey",
+                (string)status["suspendPostUri"]);
+            Assert.Equal(
+                $"{TestConstants.NotificationUrlBase}/instances/7b59154ae666471993659902ed0ba742/resume?reason={{text}}&taskHub=SampleHubVS&connection=Storage&code=mykey",
+                (string)status["resumePostUri"]);
+        }
+        
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public void CreateCheckStatusResponse_Respects_UseForwardedHost_EnvironmentVariable()
+        {
+            Environment.SetEnvironmentVariable("DurableFunctions__UseForwardedHost", "true");
```

Comment:
Rather than setting this environment variable, I think you can pass an optional parameter to `GetTestExtension()` which sets the new `options.HttpSettings.UseForwardedHost` flag you added to `HttpOptions.cs`.

Reply:
Applied suggestion